### PR TITLE
add dry-run to docker push step

### PR DESCRIPTION
**Pull request by spencerdrak**
Test Plan:

Ensure PR checks run as expected (push-dev should do a dry run), then toggle switch and push.

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: .github/workflows/push-docker.yaml [hunk 1]
```diff
@@ -11,6 +11,11 @@ on:
         required: true
         type: string
         description: The repository/name of the docker image to push, e.g., returntocorp/semgrep
+      dry-run:
+        required: true
+        type: boolean
+        description: Whether a dry-run (e.g., print tags to push) should be peformed. Actually push images if false.
```
Comment (cgdolan) [inline .github/workflows/push-docker.yaml:17]:
```suggestion
        description: Whether a dry-run (e.g., print tags to push) should be peformed. Actually push images if false.
        default: true
```

File: .github/workflows/push-docker.yaml [hunk 2]
```diff
@@ -21,6 +26,10 @@ on:
...
```
Comment (cgdolan) [inline .github/workflows/push-docker.yaml:33]:
```suggestion
        description: Whether a dry-run (e.g., print tags to push) should be peformed. Actually push images if false.
        default: true
```

**Review by cgdolan — APPROVED** [review]

**Review by cgdolan — APPROVED** [review]

**Comment by aryx** [comment]
Nice test plan:! Thanks.

File: .github/workflows/push-docker.yaml [hunk 3]
```diff
@@ -21,6 +27,11 @@ on:
         required: true
         type: string
         description: The repository/name of the docker image to push, e.g., returntocorp/semgrep
+      dry-run:
```
Comment (aryx) [inline .github/workflows/push-docker.yaml:30]: This is so ugly. Why do we need to repeat the same thing for workflow_dispatch and worflow_call.
This yaml thing is retarded.

**Review by aryx — APPROVED** [review]

**Comment by aryx** [comment]
Hmm actually I'm not sure I understand the test plan.
Did you go to https://github.com/returntocorp/semgrep/actions and then run some workflow manually with Dry run set to false?

File: .github/workflows/push-docker.yaml (hunk 3 above)
Comment (cgdolan) [reply .github/workflows/push-docker.yaml:30]: I feel this comment falls short of our respect and inclusion values. All technologies have strengths and weaknesses and while I agree that this a weakness, I think it's inappropriate to use `retarded` as a pejorative term here. My understanding from the GitHub documentation on workflow syntax is that the `inputs` context is a child of both `workflow_dispatch` and `workflow_call` and can not be shared between them: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions.

**Review by cgdolan — APPROVED** [review]

File: .github/workflows/push-docker.yaml (hunk 3 above)
Comment (aryx) [reply .github/workflows/push-docker.yaml:30]: Ok sorry, I'll try to better hide my hate for yaml.

**Pull request merged**